fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java

```diff
@@ -63,15 +64,25 @@ class ConfigurationWrap extends Configuration {
         super(loadDefaults);
     }
 
-    public String parseRegionFromEndpoint(String endPoint) {
-        if (endPoint.contains("oss")) {
+    public String parseRegionFromEndpoint(TObjectStoreType tObjectStoreType, String endPoint) {
+        if (tObjectStoreType == TObjectStoreType.S3) {
+            return AwsHostNameUtils.parseRegionFromAwsPartitionPattern(endPoint);
+        } else if (tObjectStoreType == TObjectStoreType.OSS) {
             String[] hostSplit = endPoint.split("\\.");
             String regionId = hostSplit[0];
             if (regionId.contains("-internal")) {
                 return regionId.substring(0, regionId.length() - "-internal".length());
             } else {
                 return regionId;
             }
+        } else if (tObjectStoreType == TObjectStoreType.KS3) {
+            String[] hostSplit = endPoint.split("\\.");
+            String regionId = hostSplit[0];
+            if (regionId.contains("-internal")) {
+                return regionId.substring(4, regionId.length() - "-internal".length() - 4);
```

Comment:
What's the meaning of internal?

Reply:
for oss and ks3, it has extranet endpoint and intranet endpint, 
extranet endpoint will have the 'internal'. 
Take the OSS for example:
```
extranet endpoint: oss-cn-hangzhou.aliyuncs.com 
intranet endpoint: oss-cn-hangzhou-internal.aliyuncs.com
region-id: oss-cn-hangzhou
```
when parsing the region from endpoint, we need to remove the 'internal' here